OpenRA.Mods.RA2/Traits/CarrierParent.cs

```diff
@@ -0,0 +1,286 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("This actor can spawn actors.")]
+	public class CarrierParentInfo : BaseSpawnerParentInfo
+	{
+		[Desc("Spawn is a missile that dies and not return.")]
+		public readonly bool SpawnIsMissile = false;
+
+		[Desc("Spawn rearm delay, in ticks")]
+		public readonly int RearmTicks = 150;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self right after launching a spawned unit. (Used by V3 to make immobile.)")]
+		public readonly string LaunchingCondition = null;
+
+		[Desc("After this many ticks, we remove the condition.")]
+		public readonly int LaunchingTicks = 15;
+
+		[Desc("Pip color for the spawn count.")]
+		public readonly PipType PipType = PipType.Green;
+
+		[Desc("Instantly repair spawners when they return?")]
+		public readonly bool InstantRepair = true;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while spawned units are loaded.",
+			"Condition can stack with multiple spawns.")]
+		public readonly string LoadedCondition = null;
+
+		[Desc("Conditions to grant when specified actors are contained inside the transport.",
+			"A dictionary of [actor id]: [condition].")]
+		public readonly Dictionary<string, string> SpawnContainConditions = new Dictionary<string, string>();
```

Comment:
Thinking about it: What is the use case of this? If it is not related to the default mod we should remove it which would simplify the code much. Downstream mods can easily subclass and restore such things if needed, and having unused features sitting around isn't good practice.